01-start/03-begin.html.md

```diff
@@ -565,7 +560,7 @@ As you now have all the tools and knowledge required to be able to create the bl
 
 Congratulations! You now possess all the foundations required to be able to write amazing and powerful web applications like those already in our [Showcase](/docpad/showcase). To recap, you now know how to:
 
-- write documents in any language, markup, pre-processor, templating engine, whatever you wish, by installing the necessary plugin for it and changing the extensions of the document
+- write `render` files in any language, markup, pre-processor, templating engine, whatever you wish, by installing the necessary plugin for it and changing the extensions of the document
```

Comment:
> That would involve, for a start, fixing the init process to actually create directories with the proper names, and going through and fixing all the skeletons.

Yes, DocPad should not create directories with the old names, strange it still does. PR welcome.



> Ideally, we should also provide some help for migrating existing projects so as to erase this confusion from the face of the earth.

Not a bad idea, although I'd imagine existing projects don't care, as support for both is in there. Paving the way forward with clear docs and standards should do enough to incentivise users to rename some directories. Not sure enforced migrations or warnings will do a better job.



> My sense, especially in light of the internal class/model descriptions above, is that a "document" is actually the output of the rendering process. Thus, each render file is really the specification for a document, giving the name, location and (some of the) contents of a document; the actual document is produced in the out/ directory by a rendering process combining that with layout information. This implies that the previous name of the render/ directory, documents/ was really very wrong, so wrong that I'd suggest we prioritize getting away from it completely.

This will be very complicated to change in DocPad world, as the model names are part of the code ecosystem and APIs. So having Document and File mean anything else than their current models, would not be helpful. So far I believe we have called the out files, out files.

Ideally one day, we will move away from a fixed model extension paradigm, and instead just have a single class "Document" or whatnot, which behaviour is determined completely by it's properties and the way plugins handle those properties in events emitted from it, rather than by which model class was used to define it.